src/main/java/org/broadinstitute/hellbender/tools/exome/Sample.java

```diff
@@ -29,24 +31,12 @@
      *    it contains any {@code null} or some repeated read-group ids.
      */
     public Sample(final String id, final Collection<String> readGroups) {
+        this.id = Utils.nonNull(id);
+        Utils.nonNull(readGroups);
+        Utils.containsNoNull(readGroups, "the input read-groups collection contains null values");
 
-        if (id == null) {
-            throw new IllegalArgumentException("the id cannot be null");
-        } else if (readGroups == null) {
-            throw new IllegalArgumentException("the readGroups collection cannot be null");
-        }
-        this.id = id;
-        final Set<String> readGroupSet = new LinkedHashSet<>(readGroups.size());
-        for (final String readGroupId : readGroups) {
-            if (readGroupId == null) {
-                throw new IllegalArgumentException("the input read-groups collection contains null values");
-            } else if (!readGroupSet.add(readGroupId)) {
-                throw new IllegalArgumentException(
-                        String.format("the same read-groups '%s' is found multiple times in the read-groups collection",
-                                readGroupId));
-
-            }
-        }
+        final Set<String> readGroupSet = new LinkedHashSet<>(readGroups);
+        Utils.validateArg(readGroupSet.size() == readGroups.size(), "readGroups contains duplicates");
```

Comment:
technically, the error message used to be more informative by including the name of the read group. Can you add that back?